Cython/Compiler/ModuleNode.py

```diff
@@ -3059,10 +3059,21 @@ def generate_type_import_call(self, type, code, import_generator, error_code=Non
         else:
             code.put('sizeof(%s), ' % objstruct)
 
-        code.putln('%i); if (!%s) %s' % (
-            not type.is_external or type.is_subclassed,
-            type.typeptr_cname,
-            error_code))
+        # check_size
+        if not type.is_external or type.is_subclassed:
```

Comment:
Would be better to raise a compiler error here if `check_size` was set. Or at least warn that the value was ignored.

Reply:
I am not sure what these values mean. It is from the original code and the source of the documentation problem you pointed out.

Reply:
Check added, test needed. Suggestion for `ctypedef` stanza that will trigger the failure welcome.

Reply:
"is_external" = defined in a "cdef extern" block. False if a type is cimported from another module via a .pxd file.
"is_subclassed" = we found a subtype of the type, i.e. it is definitely not a leaf class of the hierarchy.

Reply:
Or do we want to allow this for non-extern classes? (We need to take care that the vtable doesn't change either...)

Reply:
If the type is `cimport`ed, how would we set `check_size`? Is there an example in the tests that redefines a `cimport`ed type?